src/main/kotlin/com/terraformation/backend/db/ParameterizedEnum.kt

```diff
@@ -0,0 +1,5 @@
+package com.terraformation.backend.db
+
+interface ParameterizedEnum {
```

Comment:
Thinking about this a bit more, I'm not convinced about adding a new interface instead of using the existing `EnumFromReferenceTable` interface for this.

If at some point we want this kind of parsing for something other than our jOOQ-generated enums, it'd be trivial to introduce a new interface at that point, but if we don't ever want that, then having a separate interface would just make our inheritance hierarchy more complex.